### PR TITLE
[WIP] Feature/new interface for procedure

### PR DESCRIPTION
**Pull request by markusgraube**
Implement new interface between backend and frontend

### CODE REVIEW

File: src/recipe/options.ts [hunk 1]
```diff
@@ -85,7 +85,7 @@ export interface TransitionOptions {
 }
 
 export type ConditionOptions = AndConditionOptions | TimeConditionOptions | OrConditionOptions |
-    TimeConditionOptions | StateConditionOptions | VariableConditionOptions | NotConditionOptions |
+     StateConditionOptions | VariableConditionOptions | NotConditionOptions |
```
Comment (markusgraube) [inline src/recipe/options.ts:88]: Why is TimeConditionOptions removed?

**Comment by markusgraube** [comment]
All in all, it's mainly about renaming with the proper words from the standards

**Comment by markusgraube** [comment]
@julorenz What about this PR? Should we try to integrate it or just delete it?

**Comment by julorenz** [comment]
we should try to integrate it, currently published npm alpha version is already on this level
intention was to create one version running, even with new frontend, this wasn't finished yet, thats why I didnt merge this
should we do it nevertheless?

**Comment by markusgraube** [comment]
ok, then let's wait for the frontend

**Comment by tuansondinh** [comment]
we should merge this, should we? @julorenz

**Review by julorenz — APPROVED** [review]

**Pull request merged**